contracts/FlowTransactionScheduler.cdc

```diff
@@ -0,0 +1,1440 @@
+    /// that is used for governing the protocol
+    /// This is an interface to allow for the configuration details to be updated in the future
+    access(all) struct interface SchedulerConfig {
+
+        /// maximum effort that can be used for any transaction
+        access(all) var maximumIndividualEffort: UInt64
+
+        /// slot total effort limit is the maximum effort that can be 
+        /// cumulatively allocated to one timeslot by all priorities
+        access(all) var slotTotalEffortLimit: UInt64
+
+        /// slot shared effort limit is the maximum effort 
+        /// that can be allocated to high and medium priority 
+        /// transactions combined after their exclusive effort reserves have been filled
+        access(all) var slotSharedEffortLimit: UInt64
+
+        /// priority effort reserve is the amount of effort that is 
+        /// reserved exclusively for each priority
+        access(all) var priorityEffortReserve: {Priority: UInt64}
+
+        /// priority effort limit is the maximum cumulative effort per priority in a timeslot
+        access(all) var priorityEffortLimit: {Priority: UInt64}
+
+        /// minimum execution effort is the minimum effort that can be 
+        /// used for any transaction
+        access(all) var minimumExecutionEffort: UInt64
+
+        /// max data size is the maximum data size that can be stored for a transaction
+        access(all) var maxDataSizeMB: UFix64
+
+        /// priority fee multipliers are values we use to calculate the added 
+        /// processing fee for each priority
+        access(all) var priorityFeeMultipliers: {Priority: UFix64}
+
+        /// refund multiplier is the portion of the fees that are refunded when any transaction is cancelled
+        access(all) var refundMultiplier: UFix64
+
+        /// canceledTransactionsLimit is the maximum number of canceled transactions
+        ///to keep in the canceledTransactions array
+        access(all) var canceledTransactionsLimit: UInt
+
+        /// collectionEffortLimit is the maximum effort that can be used for all transactions in a collection
+        access(all) var collectionEffortLimit: UInt64
+
+        /// collectionTransactionsLimit is the maximum number of transactions that can be processed in a collection
+        access(all) var collectionTransactionsLimit: Int
+
+        access(all) init(
+            maximumIndividualEffort: UInt64,
+            slotSharedEffortLimit: UInt64,
+            priorityEffortReserve: {Priority: UInt64},
+            priorityEffortLimit: {Priority: UInt64},
+            minimumExecutionEffort: UInt64,
+            maxDataSizeMB: UFix64,
+            priorityFeeMultipliers: {Priority: UFix64},
+            refundMultiplier: UFix64,
+            canceledTransactionsLimit: UInt,
+            collectionEffortLimit: UInt64,
+            collectionTransactionsLimit: Int
+        ) {
+            pre {
+                refundMultiplier >= 0.0 && refundMultiplier <= 1.0:
```

Comment:
I believe that alowing the refundMultiplier be set to 1.0 will cause issues in the refund function, since it will try to withdraw 0

Reply:
withdrawing zero is allowed, so I don't think this is a problem